AzureIoTHub.agent.lib.nut

```diff
@@ -629,513 +604,842 @@ class AzureIoTHub {
 
     //------------------------------------------------------------------------------
 
-    // Client transport class
-    // NOTE: This class differs from the original Node.js SDK
-    ClientAMQP = class {
-
-        _config = null;
-        _connection = null;
-        _sessions = null;
-        _senders = null;
-        _receivers = null;
-        _transfers = null;
-        _handlers = null;
-        _msgQueue = null;
-
-        _connecting = false;
-        _senderTokenError = false;
-        _receiverTokenError = false;
-        _debug = false;
+    Client = class {
 
-        constructor(config) {
-            _config = config;
-            _resetConnectionTables();
-        }
+        static DEFAULT_CLIENT_OPTIONS = {
+            "qos" : 0,
+            // Timeout for RetrieveTwin and UpdateTwin requests (sec)
+            "timeout" : 10,
+            // Maximum amount of parallel UpdateTwin requests
+            "twinUpdParallelReqs" : 3,
+            // Maximum amount of parallel SendMessage requests
+            "sendMsgParallelReqs" : 3
+        };
 
-        // done has one param (error)
-        // onDisconnect has one param (error)
-        function connect(done = null, onDisconnect = null) {
-            // set connecting flag
-            _connecting = true;
-
-            // set callbacks
-            _handlers.onConnected <- done;
-            _handlers.onDisconnect <- onDisconnect;
-
-            // TODO: create a state machine to clean up tracking connection status
-            // Don't open a connection if one is already open
-            if( _isOpen(_connection) ) {
-                if ( _isOpen(_sessions.auth) ) {
-                    if ( _isOpen(_receivers.auth) ) {
-                        if ( _isOpen(_senders.auth) ) {
-                            if ( _isOpen(_sessions.event) ) {
-                                if ( _isOpen(_senders.event) ) {
-                                    _connectionCallback(null);
-                                } else {
-                                    _openEventSender();
-                                }
-                            } else {
-                                _openEventSession();
-                            }
-                        } else {
-                            _openAuthSender();
-                        }
-                    } else {
-                        _openAuthReceiver();
-                    }
-                } else {
-                    _openAuthSession();
-                }
-            } else {
-                _connection = amqp.openconnection("amqps://" + _config.host, _amqpConnectionStatusHandler.bindenv(this));
+        static S_DISCONNECTED       = 1;
```

Comment:
please move them to constants with a prefix

Reply:
AZURE_IOT_CLIENT_STATE_...

Reply:
Done